src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -27,6 +27,18 @@ public static bool IsProcessRunning(int processId)
         /// <returns>true if the process is running; otherwise, false.</returns>
         public static bool IsProcessRunning(int processId, string machineName)
         {
+            //Performance optimization for the local machine: 
+            //First try to OpenProcess by id, if valid handle is returned, the process is definitely running
+            //Otherwise enumerate all processes and compare ids
+            if (!IsRemoteMachine(machineName))
+            {
+                SafeProcessHandle processHandle = Interop.Kernel32.OpenProcess(Interop.Advapi32.ProcessOptions.PROCESS_QUERY_INFORMATION, false, processId);
+                if (!processHandle.IsInvalid)
+                {
+                    return true;
+                }
+            }
+
             return IsProcessRunning(processId, GetProcessIds(machineName));
```

Comment:
Now that you've removed one of the call sites to this `IsProcessRunning(int,int[])` overload, is it being used from anywhere other than here?  That function isn't really doing anything to determine if a process is running, it's just checking whether an integer is in an array of integers, with Array.IndexOf... we might as well just replace the call with direct usage of Array.IndexOf here, if no one else is using that other method.